core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -72,6 +72,16 @@ sealed abstract class ConfigReaderFailure {
   def path: Option[String]
 
   /**
+   * A human-readable description of the failure.
+   */
+  def description: String
+
+  /**
+   * A human-readable description of the failure, along with its location.
+   */
+  def descriptionWithLocation: String = location.fold(description)(_.description + " " + description)
```

Comment:
I preferred if this was a private function local to `ConfigReaderException`. In my view at least, `ConfigReaderFailure` should be as decoupled from presentation as possible, while all logic for turning "failure" domain models into nice error messages should be in `ConfigReaderException` whenever possible. What do you think?